File: server/src/lib/apiResponse.js
export function success(res, data, statusCode = 200) {
  return res.status(statusCode).json({ success: true, data, error: null });
}

export function error(res, message, statusCode = 400, extra = {}) {
  if (extra.retryAfter) {
    res.setHeader('Retry-After', String(extra.retryAfter));
  }
  return res.status(statusCode).json({ success: false, data: null, error: message });
}

File: pg_mobile/lib/core/network/api_response.dart
class ApiResponse<T> {
  const ApiResponse({
    required this.success,
    this.data,
    this.error,
  });

  factory ApiResponse.fromJson(
    Map<String, dynamic> json,
    T Function(Object? json) fromJsonT,
  ) {
    return ApiResponse(
      success: json['success'] as bool? ?? false,
      data: json['data'] != null ? fromJsonT(json['data']) : null,
      error: json['error'] as String?,
    );
  }

  final bool success;
  final T? data;
  final String? error;
}
